### ISSUE TITLE
[Feature]: Enable turning pii masking on/off per request

### ISSUE THREAD

**Issue by krrishdholakia** [BerriAI/litellm#2003]
### The Feature

Allow app owner to turn on/off pii masking per request

### Motivation, pitch

user request

**Comment by ghost** [comment]
picking this up today

**Comment by ghost** [comment]
## Thinking 

This will probably look like caching: 

```
import os
from openai import OpenAI

client = OpenAI(
    # This is the default and can be omitted
    api_key=os.environ.get("OPENAI_API_KEY"),
        base_url="http://0.0.0.0:8000"
)

chat_completion = client.chat.completions.create(
    messages=[
        {
            "role": "user",
            "content": "Say this is a test",
        }
    ],
    model="gpt-3.5-turbo",
    pii={"no-pii": True}
)
```

**Comment by ghost** [comment]
Thinking further, we'll probably need to expose similar controls for content moderation. 

Would it make more sense to rename the variable as moderation or content_safety? 

```python
import os
from openai import OpenAI

client = OpenAI(
    # This is the default and can be omitted
    api_key=os.environ.get("OPENAI_API_KEY"),
        base_url="http://0.0.0.0:8000"
)

chat_completion = client.chat.completions.create(
    messages=[
        {
            "role": "user",
            "content": "Say this is a test",
        }
    ],
    model="gpt-3.5-turbo",
    content_safety={"no-pii": True} # extend in the future to be {"no-pii": True, "no-moderation": True}
)
```

**Comment by ghost** [comment]
Merged into main, live in v`1.25.2`+
